geomstats/learning/mdm.py

```diff
@@ -0,0 +1,99 @@
+"""The MDM classifier on manifolds."""
+
+import geomstats.backend as gs
+from geomstats.learning.frechet_mean import FrechetMean
+
+
+class RiemannianMinimumDistanceToMeanClassifier():
+    """
+    Classifier implementing the MDM scheme on manifolds.
+
+    Parameters
+    ----------
+    riemannian_metric : string or callable.
+        The distance metric to use.
+
+    Attributes
+    ----------
+    G: array-like, shape=[n_classes, n_features] if point_type='vector'
+                   shape=[n_classes, n_features, n_features]
+                    if point_type='matrix'
+       Frechet means of each class.
+    """
+
+    def __init__(
+            self,
+            riemannian_metric,
+            point_type='matrix'):
+        self.riemannian_metric = riemannian_metric
+        self.point_type = point_type
+        self.G = None
+
+    def fit(self, X, Y):
+        """
+        Compute Frechet mean of each class.
+
+        :param X: array-like, shape=[n_samples, n_features]
+                              if point_type='vector'
+                              shape=[n_samples, n_features, n_features]
+                              if point_type='matrix'
+                  Training data, where n_samples is the number of samples
+                  and n_features is the number of features.
+        :param Y: array-like, shape=[n_samples, n_classes]
+                  Training labels, where n_classes is the number of classes.
+        """
+        n_classes = Y.shape[-1]
```

Comment:
`n_classes` could be an attribute of the class, see:
https://github.com/geomstats/geomstats/blob/13bedc24be53179d68954d464d207a7ff56087a7/geomstats/learning/kmeans.py#L47

for KMeans.

Reply:
Done.